RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -118,74 +114,56 @@ private string BuildLocalCopyAssignment()
                 + _localCopyVariableName + " = " + _target.IdentifierName;
         }
 
-        private IEnumerable<string> GetIdentifierNamesAccessibleToProcedureContext(RuleContext ruleContext)
+        private IEnumerable<string> GetIdentifierNamesAccessibleToProcedureContext()
         {
-            var allIdentifiers = new HashSet<string>();
-
-            var blocks = QuickFixHelper.GetBlockStmtContextsForContext(ruleContext);
-
-            var blockStmtIdentifiers = GetIdentifierNames(blocks);
-            allIdentifiers.UnionWith(blockStmtIdentifiers);
-
-            var args = QuickFixHelper.GetArgContextsForContext(ruleContext);
-
-            var potentiallyUnreferencedParameters = GetIdentifierNames(args);
-            allIdentifiers.UnionWith(potentiallyUnreferencedParameters);
-
-            //TODO: add module and global scope variableNames to the list.
-
-            return allIdentifiers.ToArray();
+            var allSameProcedureDeclarations = _parserState.AllUserDeclarations
+                    .Where(item => item.ParentScope == _target.ParentScope)
+                    .ToList();
+
+            var sameModuleDeclarations = _parserState.AllUserDeclarations
+                    .Where(item => item.ComponentName == _target.ComponentName
+                    && !IsDeclaredInMethodOrProperty(item.ParentDeclaration.Context))
+                    .ToList();
+
+            var allGloballyAccessibleDeclarations = _parserState.AllUserDeclarations
+                .Where(item => item.ProjectName == _target.ProjectName
+                && !(item.ParentScopeDeclaration is ClassModuleDeclaration)
+                && (item.Accessibility == Accessibility.Public 
+                    || ((item.Accessibility == Accessibility.Implicit) 
+                        && (item.ParentScopeDeclaration is ProceduralModuleDeclaration))))
+                .ToList();
+
+            var accessibleIdentifierNames = new List<string>();
+            accessibleIdentifierNames.AddRange(allSameProcedureDeclarations.Select(d => d.IdentifierName));
+            accessibleIdentifierNames.AddRange(sameModuleDeclarations.Select(d => d.IdentifierName));
+            accessibleIdentifierNames.AddRange(allGloballyAccessibleDeclarations.Select(d => d.IdentifierName));
+
+            return accessibleIdentifierNames.Distinct();
         }
 
-        private IEnumerable<string> GetIdentifierNames(IEnumerable<RuleContext> ruleContexts)
+        private bool IsDeclaredInMethodOrProperty(RuleContext context)
```

Comment:
A map of member types would look cleaner - but that's good enough; soon we'll have the context types share an interface and that annoyance won't be needed anymore. Well done!